ocs_ci/ocs/longevity_stage2.py

```diff
@@ -0,0 +1,588 @@
+
+    for line in logs:
+        logger.info(line)
+        if "Scheduled" in line:
+            scheduled_time = int(line.split()[0][:-1])
+        elif "Created" in line:
+            created_time = int(line.split()[0][:-1])
+            creation_time = scheduled_time - created_time
+            logger.info(f"POD number {pod_no} was created in {creation_time} seconds.")
+            if creation_time > accepted_creation_time:
+                raise ex.PerformanceException(
+                    f"POD creation time is {creation_time} and is greater than "
+                    f"{accepted_creation_time} seconds."
+                )
+            pod_no -= 1
+
+
+def measure_pvc_creation_time(interface, pvc_objs, start_time):
+    """
+    Measures and Logs PVC Creation Time of all PVCs.
+
+    Args:
+        interface (str) : an interface (RBD or CephFS) to run on.
+        pvc_objs (list) : List of PVC objects for which we have to measure the time.
+        start_time (str) : Formatted time from which and on to search the relevant logs.
+
+    Raises:
+        PerformanceException : Raises an exception if PVC creation time is greater than the accepted time.
+
+    Logs:
+        PVC Creation Time of all the PVCs.
+
+    """
+    accepted_creation_time = 1
+    for pvc_obj in pvc_objs:
+        creation_time = performance_lib.measure_pvc_creation_time(
+            interface, pvc_obj.name, start_time
+        )
+
+        logger.info(f"PVC {pvc_obj.name} was created in {creation_time} seconds.")
+        if creation_time > accepted_creation_time:
+            raise ex.PerformanceException(
+                f"PVC {pvc_obj.name} creation time is {creation_time} and is greater than "
+                f"{accepted_creation_time} seconds."
+            )
+
+
+def measure_pvc_deletion_time(interface, pvc_objs):
+    """
+    Measures and Logs PVC Deletion Time of all PVCs.
+
+    Args:
+        interface (str) : an interface (RBD or CephFS) to run on.
+        pvc_objs (list) : List of PVC objects for which we have to measure the time.
+
+    Raises:
+        PerformanceException : Raises an exception if PVC deletion time is greater than the accepted time.
+
+    Logs:
+        PVC Creation Time of all the PVCs.
+
+    """
+    accepted_deletion_time = 2 if interface == constants.CEPHFILESYSTEM else 1
+    num_of_pvcs = len(pvc_objs)
+
+    for pvc_no in range(num_of_pvcs):
+        pvc = pvc_objs[pvc_no]
+        pvc_name = pvc.name
+        pvc.ocp.wait_for_delete(resource_name=pvc_name)
+        helpers.validate_pv_delete(pvc.backed_pv)
+        deletion_time = helpers.measure_pvc_deletion_time(interface, pvc.backed_pv)
+        logger.info(f"PVC {pvc_name} was deleted in {deletion_time} seconds.")
+        if deletion_time > accepted_deletion_time:
+            raise ex.PerformanceException(
+                f"PVC {pvc_name} deletion time is {deletion_time} and is greater than "
+                f"{accepted_deletion_time} seconds."
+            )
```

Comment:
this loop is going to consume lot of memory if there are lot of PVCs in the testing code. 
Better get the list of backend PVCs and call measure_pvc_creation_time_bulk which will gather logs one time and get returns data of all the requested PVCs

Reply:
I have updated the code and used measure_pv_deletion_time_bulk. But it is still taking more than 10 seconds. The details of the job can be found here - https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-deploy-ocs-cluster/12566/console